lib/src/Pos/Util/UserSecret.hs

```diff
@@ -305,3 +341,76 @@ writeRaw u = do
 -- | Helper for taking shared lock on file
 takeReadLock :: MonadIO m => FilePath -> IO a -> m a
 takeReadLock path = liftIO . withFileLock (lockFilePath path) Shared . const
+
+----------------------------------------------------------------------------
+-- Legacy
+----------------------------------------------------------------------------
```

Comment:
1) Why do you need to support it still?
2) If you intended to use it for legacy support (decoding old Daedalus keyfiles?...), then where is the logic of transforming legacy `UserSecret0` to new `UserSecret`?

Reply:
> legacy support (decoding old Daedalus keyfiles?...)

Yes, exactly.

> where is the logic of transforming legacy UserSecret0 to new UserSecret?

I think we can put it into `ariadne` repository to decrease diff between `input-output-hk/cardano-sl` and `serokell/cardano-sl`.
(OTOH, we don't want to put new `UserSecret` into `ariadne` repository, because it would require some unpleasant changes, probably extending `ReaderT` environment)